application-analytics-default/src/main/java/com/xwiki/analytics/internal/configuration/DefaultAnalyticsConfiguration.java

```diff
@@ -68,6 +68,12 @@ public String getAuthenticationToken()
         return getProperty("authToken", "");
     }
 
+    @Override
+    public String getTrackingCode()
+    {
+        return getProperty("trackingCode", "");
+    }
+
     private <T> T getProperty(String key, T defaultValue)
     {
         T value = this.configDocument.getProperty(key, defaultValue);
```

Comment:
I know it's not part of this PR, but below you are throwing a RuntimeException, which is not correct (just noticed). For the `this.configDocument.getProperty` you have a default value, therefore `value == null || value.equals(defaultValue)` will never be false.  So, your `getProperty` method is not needed, you can have directly:
```
    @Override
    public String getTrackingCode()
    {
        return this.configDocument.getProperty("trackingCode", "");
    }
```
I think you did it because of https://github.com/xwikisas/application-collabora/blob/main/application-collabora-default/src/main/java/com/xwiki/collabora/internal/configuration/DefaultCollaboraConfiguration.java#L71 , where I didn't wanted to have a default value, but to know when the property was not defined, which is not your case in here